src/components/TableDialog.vue

```diff
@@ -139,6 +185,22 @@ export default defineComponent({
     },
   },
   setup(props, { emit }) {
+    // Stepper control
+    const step: Ref<number> = ref(1);
+    const rowSample: Ref<Array<{}>> = ref([]);
+    const headers = computed(() => {
+      if (rowSample.value.length === 0) {
+        return [];
+      }
+
+      const keys = Object.keys(rowSample.value[0]);
+
+      return keys.map((key) => ({
+        text: key,
+        value: key,
+      }));
+    });
```

Comment:
I think this looks just a bit cleaner:
```suggestion
    const headers = computed(() => {
      const keys = Object.keys(rowSample.value[0] || {});
      return keys.map((key) => ({
        text: key,
        value: key,
      }));
    });
```